lightly/api/upload.py

```diff
@@ -187,22 +79,22 @@ def _upload_single_image(image,
     # upload sample with metadata
     sample_upload_success = True
 
+    api_client = get_api_client(token=token)
+    samples_api = SamplesApi(api_client=api_client)
+
     try:
-        sample_id = routes.users.datasets.samples.post(
-            basename, thumbname, metadata, dataset_id, token
-        )
-    except RuntimeError as e:
-        sample_upload_success = False
-        raise ValueError
+        body = SampleCreateRequest(file_name=basename, thumb_name=thumbname, meta_data=metadata)
+        sample_id = samples_api.create_sample_by_dataset_id(body=body, dataset_id=dataset_id).id
+    except RuntimeError:
+        raise ValueError("Creating the sampling in the web platform failed.")
```

Comment:
Creating the "sample" failed. And you shouldn't use a value error for this. IMO you can remove the try-except and check the status code of the response instead.

Reply:
Solved it with a try-except around the whole upload_single_image(), as discussed in slack.